packages/fxa-auth-server/lib/routes/recovery-key.js

```diff
@@ -249,6 +260,86 @@ module.exports = (log, db, Password, verifierVersion, customs, mailer) => {
         return db.recoveryKeyExists(uid);
       },
     },
+    {
+      method: 'GET',
+      path: '/recoveryKey/storageHint',
+      options: {
+        ...RECOVERY_KEY_DOCS.RECOVERYKEY_HINT_GET,
+        // The storageHint can be (or will be) retrieved during password reset with account recovery key
+        // (after following password reset link) or when authenticated (from /session)
+        // TODO: verify if we can allow to auth strategies
+        auth: {
+          strategy: ['accountResetToken', 'sessionToken'],
+        },
+      },
+      handler: async function (request) {
+        log.begin('getRecoveryKeyHint', request);
+
+        const { uid } = request.auth.credentials;
+
+        await customs.checkAuthenticated(request, uid, 'getRecoveryKeyHint');
+
+        const { storageHint } = await db.getRecoveryKeyHint(uid);
+
+        recordSecurityEvent('account.recovery_key_hint_viewed', {
+          db,
+          request,
+          account: { uid },
+        });
+
+        return { storageHint };
+      },
+    },
+    {
+      method: 'POST',
+      path: '/recoveryKey/storageHint',
```

Comment:
I didn't see any screens where a user updated their storage hint? I'm not sure of a scenario where they would need to do that, maybe we can leave this part out.

Reply:
We don't have a screen for that yet, but will be adding one in https://mozilla-hub.atlassian.net/browse/FXA-7238. The storage hint step is executed separately from the key creation (key is already generated and stored when the user reaches this step of the user flow), so we need a method for to add/update a storageHint to an existing key. 

I can see though that we don't need a stored procedure for this. I'll update!